src/Scrambler.js

```diff
@@ -20,30 +33,29 @@ export default class Scrambler {
     this.frameIndex = 0;
   }
 
-  scramble(text, onScramble, option = null) {
-    if (option?.characters) {
-      this.characters = [...option.characters];
-    } else {
-      this.characters = [...Scrambler.CHARACTERS.DEFAULT];
+  scramble(text, onScramble, options = null) {
+    if (options) {
+      if (options.delay) this.delay = options.delay;
+      if (options.maxCounter) this.maxCounter = options.maxCounter;
```

Comment:
How about adding validation and default value assignment for the new `maxCounter` and `delay` inputs? This could help ensure that if a user inputs a negative or very large value, a more reasonable default value is used.

Reply:
Just tested the input validation logic I suggested above for `delay` and it produces inconsistent results. A `delay`  of 3.4 seems to create a slower scrambling effect than a `delay` of 4, for example. So it may be a good idea to simply stick to whole numbers other than 0, with input validation looking something like: `if (options.delay !== undefined) this.delay = Math.round(options.delay) || 1`.

Reply:
Thank you for thoroughly reviewing the input values.

Since we’re introducing a new interface that users will interact with, I want to proceed carefully. From my experience, once an interface is set, it’s hard to change.

First, I think we could improve the current parameter names to make them more intuitive for users. For example:

- `delay` -> `speed` (where a higher number means faster speed)
- `maxCounter` -> `shuffleCount` (since it represents the number of times the characters are shuffled)

Additionally, it’s important to provide clear input guidelines. Without them, users might be confused.

Based on my tests, I believe setting `speed` as an integer between 1 and 10 would cover most use cases. As for `shuffleCount`, I agree that allowing any positive integer makes sense since some users may want unlimited shuffling.

What do you think?